### ISSUE TITLE
ERROR: Could not build wheels for beanmachine, which is required to install pyproject.toml-based projects

### ISSUE THREAD

**Issue by himaghna** [facebookresearch/beanmachine#1432]
### Issue Description
Failed to install beanmachine using the recommended installation instruction (using pip)

### Steps to Reproduce
```python -m pip install beanmachine``` in empty conda environment fails with
```
clang -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O2 -Wall -fPIC -O2 -isystem /Users/himaghnabhattacharjee/miniforge3/envs/bean_env/include -arch arm64 -fPIC -O2 -isystem /Users/himaghnabhattacharjee/miniforge3/envs/bean_env/include -arch arm64 -I/private/var/folders/2l/fl3mw7qs5vzgjlt5c8rxl63r0000gn/T/pip-install-1aed5lre/beanmachine_b2b00e5097db4f93b03e8c806cebec31/src -I/Users/himaghnabhattacharjee/miniforge3/envs/bean_env/include -I/Users/himaghnabhattacharjee/miniforge3/envs/bean_env/include/eigen3 -I/Users/himaghnabhattacharjee/miniforge3/envs/bean_env/include -I/Users/himaghnabhattacharjee/miniforge3/envs/bean_env/include/boost -I/private/var/folders/2l/fl3mw7qs5vzgjlt5c8rxl63r0000gn/T/pip-build-env-z00er72y/overlay/lib/python3.10/site-packages/pybind11/include -I/Users/himaghnabhattacharjee/miniforge3/envs/bean_env/include/python3.10 -c src/beanmachine/graph/cavi.cpp -o build/temp.macosx-11.0-arm64-cpython-310/src/beanmachine/graph/cavi.o -std=c++17 -mmacosx-version-min=10.14 -fvisibility=hidden -g0 -stdlib=libc++ -std=c++14 -Werror
      In file included from src/beanmachine/graph/cavi.cpp:8:
      In file included from /private/var/folders/2l/fl3mw7qs5vzgjlt5c8rxl63r0000gn/T/pip-install-1aed5lre/beanmachine_b2b00e5097db4f93b03e8c806cebec31/src/beanmachine/graph/distribution/distribution.h:10:
      /private/var/folders/2l/fl3mw7qs5vzgjlt5c8rxl63r0000gn/T/pip-install-1aed5lre/beanmachine_b2b00e5097db4f93b03e8c806cebec31/src/beanmachine/graph/graph.h:9:10: fatal error: 'Eigen/Dense' file not found
      #include <Eigen/Dense>
               ^~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
```

### Expected Behavior
Successful installation

### System Info
- MacOS 12.2.1 (M1 Chip)
- Empty conda environment with
  - pip 22.0.4
  - python 3.10.4
  - No PyTorch

### Additional Context
Complete trace:
[beanmachine_trace.txt](https://github.com/facebookresearch/beanmachine/files/8611876/beanmachine_trace.txt)

**Comment by horizon-blue** [comment]
Thanks for providing the full trace, @himaghna. You're seeing this error because Bean Machine does not ship with prebuild binaries for M1 macbook yet. As a result, when you run `pip install beanmachine`, you're essentially building Bean Machine from source. We are working on adding M1 binaries to our release (https://github.com/facebookresearch/beanmachine/pull/1419). In the mean time, you can try installing Bean Machine from source by following the instruction in the README: https://github.com/facebookresearch/beanmachine#install-from-source

**Comment by himaghna** [comment]
> Thanks for providing the full trace, @himaghna. You're seeing this error because Bean Machine does not ship with prebuild binaries for M1 macbook yet. As a result, when you run `pip install beanmachine`, you're essentially building Bean Machine from source. We are working on adding M1 binaries to our release (#1419). In the mean time, you can try installing Bean Machine from source by following the instruction in the README: https://github.com/facebookresearch/beanmachine#install-from-source

Thanks!! @horizon-blue

**Comment by horizon-blue** [comment]
Closing this issue because binary wheels for M1 Apple Silicon has been added in https://github.com/facebookresearch/beanmachine/pull/1419 and is included in [v0.1.2 release](https://github.com/facebookresearch/beanmachine/releases/tag/v0.1.2)
